bitsail-connectors/connector-hbase/src/main/java/com/bytedance/bitsail/connector/hbase/auth/KerberosFileHandler.java

```diff
@@ -0,0 +1,183 @@
+    boolean useBase64Content = MapUtils.getBooleanValue(hbaseConfigMap, KEY_USE_BASE64_CONTENT);
+    if (useBase64Content) {
+      hbaseConfigMap.put(KEY_PRINCIPAL_FILE, KEY_KEYTAB_CONTENT_TMP_FILEPATH);
+      hbaseConfigMap.put(KEY_JAVA_SECURITY_KRB5_CONF, KEY_KRB5_CONTENT_TMP_FILEPATH);
+      hbaseConfigMap.put(KEY_USE_LOCAL_FILE, true);
+    }
+  }
+
+  /**
+   * Store kerberos authentication content into local files.
+   * @param hbaseConfigMap User defined hbase configuration.
+   */
+  public static void loadConfContent(Map<String, Object> hbaseConfigMap) throws IOException {
+    if (hbaseConfLoaded) {
+      LOG.info("Hbase kerberos file already loaded.");
+    } else {
+      handleAuthenticationBase64Content(hbaseConfigMap);
+    }
+  }
+
+  /**
+   * Return the principal file name defined in configuration.
+   * @param config HBase configuration.
+   * @return Principal file name.
+   */
+  public static String getPrincipalFileName(Map<String, Object> config) {
+    String fileName = MapUtils.getString(config, "principalFile");
+    if (org.apache.commons.lang.StringUtils.isEmpty(fileName)) {
+      throw new RuntimeException("[principalFile] must be defined!");
+    }
+
+    boolean useLocalFile = MapUtils.getBooleanValue(config, KEY_USE_LOCAL_FILE);
+    if (useLocalFile) {
+      return fileName;
+    } else {
+      throw new RuntimeException("Only support local file to authenticate.");
+    }
+  }
+
+  /**
+   * Load krb.conf file.
+   * @param kerberosConfig HBase configuration.
+   */
+  public static void loadKrb5Conf(Map<String, Object> kerberosConfig) {
+    String krb5FilePath = MapUtils.getString(kerberosConfig, KEY_JAVA_SECURITY_KRB5_CONF);
+    if (StringUtils.isEmpty(krb5FilePath)) {
+      LOG.info("krb5 file is empty, will use default file");
+      return;
+    }
+
+    krb5FilePath = loadFile(kerberosConfig, krb5FilePath);
+    kerberosConfig.put(KEY_JAVA_SECURITY_KRB5_CONF, krb5FilePath);
+  }
+
+  /**
+   * kerberosConfig:
+   * <p>{
+   * "principalFile":"keytab.keytab",
+   * "remoteDir":"/home/admin",
+   * "sftpConf":{
+   * "path" : "/home/admin",
+   * "password" : "******",
+   * "port" : "22",
+   * "auth" : "1",
+   * "host" : "127.0.0.1",
+   * "username" : "admin"
+   * }
+   * }</p>
+   */
+  public static String loadFile(Map<String, Object> kerberosConfig, String filePath) {
+    boolean useLocalFile = MapUtils.getBooleanValue(kerberosConfig, KEY_USE_LOCAL_FILE);
+    if (useLocalFile) {
+      LOG.info("Will use local file:{}", filePath);
+      File file = new File(filePath);
+      if (file.exists()) {
+        if (file.isDirectory()) {
+          throw new RuntimeException("The path is a directory:" + filePath);
+        }
+      } else {
+        throw new RuntimeException("The path does not exist:" + filePath);
+      }
+    } else {
+      throw new RuntimeException("Only support local file to authenticate.");
+    }
+    return filePath;
```

Comment:
Using [Return Early pattern](https://medium.com/swlh/return-early-pattern-3d18a41bba8) instead?

Reply:
This entire file is borrowed from the legacy hbase connector. Since this function is short, it's okay to me.